src/adera/Machines/Rocket.h

```diff
@@ -38,6 +38,8 @@ class SysMachineRocket :
 {
 public:
 
+    static const std::string smc_name;
```

Comment:
static inline const std::string smc_name = "Rocket";

Though, that being said for string constants that are used in most situations for comparisons should probably be std::string_view.

Functions that accept some kind of string, but don't need to transfer ownership, should accept a std::string_view.

If ownership needs to be transfered, you can either:

take an std::string by value, and std::move() it into place.
take an std::string by both const& and &&, and then where the caller has an std::string_view, and not an std::string, have the caller construct the std::string that's needed.
If ownership only transfers *sometimes* then also provide an std::string_view overload that only constructs a std::string when absolutely necessary. E.g. for functions that do a lookup against an std::map or similar, use the C++17 (14?) thing "std::less<>" to allow "transparent" comparisons, and then if you need to insert a new item into the map, that's when you'd allocate the string.

Reply:
To provide some more background information on this.

complex objects, like std::string incur an application startup initialization cost that involves a few mutexes and some runtime operations.

On the other hand, doing

"static constexpr std::string_view smc_name = "MyString";

Puts the string data into the read-only-data section of the binary, which incurs literally zero initialization cost, and allows the machine code to just refer to the address in the ROData section directly (with a single fixup by the runtime linker for position-independent-code related sillyness).

It also enables constant propagation to be more effective, and as a result can allow for same-code-folding to pick up on identical code where the read-only-data string is used that it might not be able to if you were using an std::string.

Reply:
How well does this work across shared libraries? In a previous comment, you mentioned "comparison of the pointer values," and it really seems like this wouldn't work if the variable is defined in multiple places.

Reply:
Like many things, it depends on many things :-)

Inline variable is going to be a weak symbol in the shared library, so the platform runtime linker might consolidate them all to a single memory address.

Nevertheless, you can't rely on comparing only pointers to determine if two strings have the same contents. That's great for positive identification, but can result in false-negatives.